for/joomla.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: page
+title: Codeception for Joomla
+hero: joomla_hero.html
+sidebar: |
+
+  ## Codeception Tests
+
+  * Combine **all testing levels** (acceptance, functional, unit)
+  * **Scenario-Driven**:  described in easy to get PHP DSL
+  * Provide common actions and assertions
+
+  ## Reference
+
+  * [Joomla Browser](http://codeception.com/addons#joomla-browserhttpsgithubcomjoomla-projectsjoomla-browser) 
+  * [Demo Tested Extension](https://github.com/joomla-extensions/weblinks#tests)
+
+---
+
+## Setup
+
+With Codeception you can create [Unit](http://codeception.com/docs/05-UnitTests), [Functional](http://codeception.com/docs/04-FunctionalTests) and [Api](http://codeception.com/docs/10-WebServices) tests, but the present documentation will covers only how to start with  [Acceptance](http://codeception.com/docs/03-AcceptanceTests) tests in Joomla and Joomla Extensions in order to  check the UI and work of a system as a whole. 
+
+### Project Setup
+Create a Joomla website in your local webserver.
+
+Go to the folder where your website is located and run:
+
+```bash
+composer require codeception/codeception --dev
+```
+
+Codeception should be installed globally for a project. To start please run
+
+```
+vendor/bin/codecept bootstrap --empty
+```
+
+which creates `codeception.yml` and `tests` directory. There are no test suites in `tests` directory. 
+
+### Acceptance Testing
+
+For UI testing you will have to create first suite for acceptance tests:
+
+```
+vendor/bin/codecept generate:suite acceptance
+```
+
+This will create `acceptance.suite.yml` and `acceptance` folder inside `tests`. Acceptance tests should be configured to work with Selenium WebDriver to test application inside a real browser. 
+
+```yaml
+class_name: AcceptanceTester
+modules:
+    enabled:
+        - PhpBrowser:
+            url: http://joomla.box/codeception/
+        - \Helper\Acceptance          
+```
+
+Acceptance tests should be described in Cest format. Run code generator 
+
+```
+vendor/bin/codecept generate:cest acceptance AdminLoginCest
+```
+
+to generate very first test `tests/acceptance/AdminLoginCest.php`. Each method of a class (except `_before` and `_after`) is a test. Tests use `$I` object (instance of `AcceptanceTester` class) to perform actions on a webpage. 
+
+Modify your `tests/acceptance/AdminLoginCest.php` using the following example code:
+
+```php
+<?php
+
+class AdminLoginCest
+{
+    public function login(AcceptanceTester $I)
+    {
+        $I->amOnPage('/administrator/index.php');
+        $I->comment('Fill Username Text Field');
+        $I->fillField(['id' => 'mod-login-username'], 'admin');
```

Comment:
`['id' => 'mod-login-username']` can be simplified to `#mod-login-username` afaik, this locator type is optimized to search by id only and not to do fuzzy search